core/src/main/java/io/opencensus/stats/Aggregate.java

```diff
@@ -0,0 +1,273 @@
+     * will throw an {@code UnsupportedOperationException}.
+     *
+     * @return the aggregated bucket counts.
+     */
+    public abstract List<Long> get();
+
+    @Override
+    public final <T> T match(
+        Function<? super Sum, T> p0,
+        Function<? super Count, T> p1,
+        Function<? super Histogram, T> p2,
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p2.apply(this);
+    }
+  }
+
+  /** The range of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class Range extends Aggregate {
+
+    Range() {
+    }
+
+    static Range create(double min, double max) {
+      if (min != Double.POSITIVE_INFINITY && max != Double.NEGATIVE_INFINITY) {
+        checkArgument(min <= max, "max should be greater or equal to min.");
+      }
+      return new AutoValue_Aggregate_Range(min, max);
+    }
+
+    /**
+     * Returns the minimum of the population values.
+     *
+     * @return the minimum of the population values.
+     */
+    // TODO(songya): not v0.1, expose this method later.
+    public abstract double getMin();
+
+    /**
+     * Returns the maximum of the population values.
+     *
+     * @return the maximum of the population values.
+     */
+    // TODO(songya): not v0.1, expose this method later.
+    public abstract double getMax();
+
+    @Override
+    public final <T> T match(
+        Function<? super Sum, T> p0,
+        Function<? super Count, T> p1,
+        Function<? super Histogram, T> p2,
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p3.apply(this);
+    }
+  }
+
+  /** The mean value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class Mean extends Aggregate {
+
+    Mean() {
+    }
+
+    static Mean create(double mean) {
+      return new AutoValue_Aggregate_Mean(mean);
+    }
+
+    /**
+     * Returns the aggregated mean.
+     *
+     * @return the aggregated mean.
+     */
+    // TODO(songya): not v0.1, expose this method later.
+    abstract double get();
+
+    @Override
+    public final <T> T match(
+        Function<? super Sum, T> p0,
+        Function<? super Count, T> p1,
+        Function<? super Histogram, T> p2,
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p4.apply(this);
+    }
+  }
+
+  /** The standard deviation value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class StdDev extends Aggregate {
+
+    StdDev() {
+    }
+
+    static StdDev create(double stdDev) {
```

Comment:
I don't think `Aggregate.create()` needs to be public, since `Aggregate`s should all be created internally, and all users need is to read stats from them, i.e expose `get()` methods.